src/components/searchPage.js

```diff
@@ -1,24 +1,97 @@
+// SearchPage.js
+import React, { useState, useEffect } from "react";
 import LogoSection from "./logoSection";
 import ProductList from "./productList";
 import Header from "./header";
-import { formatProductListForSearchPage, getSummaryProductListAsync } from "../service/productService";
-import { useSelector } from "react-redux";
 import Footer from './footer';
-
-const productsToShow = await getSummaryProductListAsync();
+import ColorFilter from "./colorFilter";
+import {
+  formatProductListForSearchPage,
+  getSummaryProductListAsync,
+} from "../service/productService";
+import { useSelector } from "react-redux";
 
 function SearchPage() {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [productsToShow, setProductsToShow] = useState([]);
+  const [filteredProducts, setFilteredProducts] = useState([]);
+
   const cartElementList = useSelector((state) => state.cart.list);
   const cartProductIdList = cartElementList.map((element) => element.id);
-  const searchProductList = formatProductListForSearchPage(productsToShow, cartProductIdList);
+ 
+  // Función para filtrar por color
+const handleColorFilter = (color) => {
+  const normalizedColor = color.toLowerCase();
+  if (normalizedColor === searchTerm) {
+    // Si se hace clic en el mismo color, limpiar los filtros
+    setSearchTerm("");
+  } else {
+    setSearchTerm(normalizedColor);
+  }
+};
+
+  // Filtrar productos por término de búsqueda y/o color
+  const searchProductList = productsToShow.filter((product) => {
+    const matchesSearchTerm = product.name.toLowerCase().includes(searchTerm.toLowerCase());
+
+    return matchesSearchTerm;
+  });
+
+  // Obtener y establecer la lista de productos al cargar la página
+  useEffect(() => {
+    const fetchProducts = async () => {
+      const products = await getSummaryProductListAsync();
+      setProductsToShow(products);
+      setFilteredProducts(products); // Inicializar la lista filtrada con todos los productos
```

Comment:
Here let's keep only `setFilteredProducts` (it should be renamed to `setFullProductList` ).
This is because this useEffect is only in charge to fetch the full results we are interested. The responsablity to filter should be on other useEffect